src/read.c

```diff
@@ -2428,41 +2428,46 @@ static void            ReadQUIT (
 **              |  ';'
 */
+static Int TryReadStatement(TypSymbolSet follow)
+{
+    switch (STATE(Symbol)) {
+    case S_IDENT:     ReadCallVarAss(follow,'s'); break;
+    case S_UNBIND:    ReadUnbind(    follow    ); break;
+    case S_INFO:      ReadInfo(      follow    ); break;
+    case S_ASSERT:    ReadAssert(    follow    ); break;
+    case S_IF:        ReadIf(        follow    ); break;
+    case S_FOR:       ReadFor(       follow    ); break;
+    case S_WHILE:     ReadWhile(     follow    ); break;
+    case S_REPEAT:    ReadRepeat(    follow    ); break;
+    case S_BREAK:     ReadBreak(     follow    ); break;
+    case S_CONTINUE:  ReadContinue(  follow    ); break;
+    case S_RETURN:    ReadReturn(    follow    ); break;
+    case S_TRYNEXT:   ReadTryNext(   follow    ); break;
+    case S_ATOMIC:    ReadAtomic(    follow    ); break;
+    case S_SEMICOLON: ReadEmpty(     follow    ); break;
+    case S_QUIT:      SyntaxError("'quit;' cannot be used in this context"); break;
+    case S_QQUIT:     SyntaxError("'QUIT;' cannot be used in this context"); break;
+    case S_HELP:      SyntaxError("'?' cannot be used in this context"); break;
+    default:         return 0;
+    }
+    return 1;
+}
+
 static UInt ReadStats (
     TypSymbolSet        follow )
 {
-    UInt               nr;            /* number of statements            */
+    UInt nr = 0; // number of statements
 
-    /* read the statements                                                 */
-    nr = 0;
+    // read the statements
     while ( IS_IN( STATE(Symbol), STATBEGIN|S_SEMICOLON ) ) {
-
-        /* read a statement                                                */
-        switch (STATE(Symbol)) {
-        case S_IDENT:     ReadCallVarAss(follow,'s'); break;
-        case S_UNBIND:    ReadUnbind(    follow    ); break;
-        case S_INFO:      ReadInfo(      follow    ); break;
-        case S_ASSERT:    ReadAssert(    follow    ); break;
-        case S_IF:        ReadIf(        follow    ); break;
-        case S_FOR:       ReadFor(       follow    ); break;
-        case S_WHILE:     ReadWhile(     follow    ); break;
-        case S_REPEAT:    ReadRepeat(    follow    ); break;
-        case S_BREAK:     ReadBreak(     follow    ); break;
-        case S_CONTINUE:  ReadContinue(  follow    ); break;
-        case S_RETURN:    ReadReturn(    follow    ); break;
-        case S_TRYNEXT:   ReadTryNext(   follow    ); break;
-        case S_ATOMIC:    ReadAtomic(    follow    ); break;
-        case S_SEMICOLON: ReadEmpty(     follow    ); break;
-        case S_QUIT:      SyntaxError("'quit;' cannot be used in this context"); break;
-        case S_QQUIT:     SyntaxError("'QUIT;' cannot be used in this context"); break;
-        case S_HELP:      SyntaxError("'?' cannot be used in this context"); break;
-        default:          ReadEmpty(     follow    ); break;
+        if (!TryReadStatement(follow)) {
+            SyntaxError( "statement expected");
```

Comment:
formatting is off here

Reply:
Indeed it is -- or rather, was: I just pushed a fixed version.